### ISSUE TITLE
I wrote the TensorFlow version of your work in Keras.

### ISSUE THREAD

**Issue by ctmakro** [bstriner/keras-adversarial#6]
I wrote it yesterday, and today I came across your repo so just wanna let you know.

Here's the description: https://ctmakro.github.io/site/on_learning/fast_gan_in_keras.html
w/ code inside.

However I didn't make it into a class.

Oh and, your DCGAN implementation seems to have diversity problem(a lot of the generations are exactly the same). you should 1) single-sided label smoothing 2) add decaying noise to input 3) use batch discrimination.

**Comment by kechan** [comment]
I experienced this "diversity" problem with example_gan.py. My generated MNIST are all 1s and 7s. =(.
